From 06e0a1967293c9f9f0635dd27ee61377f1bf5a6c Mon Sep 17 00:00:00 2001
From: kongfl888 <kongfl888@outlook.com>
Date: Thu, 9 Jul 2020 11:09:55 +0800
Subject: [PATCH] patch for 19.07.3 cpu info.

# Conflicts with 19.07.1:
#	feeds/luci/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js

Signed-off-by: kongfl888 <kongfl888@outlook.com>
---
 .../modules/luci-base/root/usr/libexec/rpcd/luci   | 35 ++++++++++++++++++++++
 .../root/usr/share/rpcd/acl.d/luci-base.json       |  2 +-
 .../resources/view/status/include/10_system.js     | 30 +++++++++++++++----
 3 files changed, 61 insertions(+), 6 deletions(-)

diff --git a/feeds/luci/modules/luci-base/root/usr/libexec/rpcd/luci b/feeds/luci/modules/luci-base/root/usr/libexec/rpcd/luci
index 35b85d4..3cf2264 100644
--- feeds/luci/modules/luci-base/root/usr/libexec/rpcd/luci
+++ feeds/luci/modules/luci-base/root/usr/libexec/rpcd/luci
@@ -115,6 +115,18 @@ local methods = {
 		end
 	},
 
+	getCPUInfo = {
+		call = function()
+			local sys = require "luci.sys"
+			local cpuinfo = {}
+
+			cpuinfo.cpuinfo = sys.exec("/bin/cpuinfo")
+
+			if (cpuinfo.cpuinfo == nil) or (cpuinfo.cpuinfo == "") then
+				cpuinfo.cpuinfo = "ARMv8 Cortex-A53 / Rockchip RK3328 / 4 Processors"
+			end
+
+			return cpuinfo
+		end
+	},
+
 	getUSBDevices = {
 		call = function()
 			local fs     = require "nixio.fs"
diff --git a/feeds/luci/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json b/feeds/luci/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
index 8b8481b..d555a88 100644
--- feeds/luci/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
+++ feeds/luci/modules/luci-base/root/usr/share/rpcd/acl.d/luci-base.json
@@ -62,7 +62,7 @@
 			"ubus": {
 				"file": [ "list", "read", "stat" ],
 				"iwinfo": [ "assoclist", "freqlist", "txpowerlist", "countrylist" ],
-				"luci": [ "getConntrackList", "getInitList", "getLocaltime", "getProcessList", "getRealtimeStats", "getTimezones", "getLEDs", "getUSBDevices", "getSwconfigFeatures", "getSwconfigPortState", "getBlockDevices", "getMountPoints" ],
+				"luci": [ "getConntrackList", "getInitList", "getLocaltime", "getProcessList", "getRealtimeStats", "getTimezones", "getLEDs", "getCPUInfo", "getUSBDevices", "getSwconfigFeatures", "getSwconfigPortState", "getBlockDevices", "getMountPoints" ],
 				"luci-rpc": [ "getBoardJSON", "getDHCPLeases", "getDSLStatus", "getDUIDHints", "getHostHints", "getNetworkDevices", "getWirelessDevices" ],
 				"network.interface": [ "dump" ],
 				"network.rrdns": [ "lookup" ],
diff --git a/feeds/luci/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js b/feeds/luci/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
index 942b2dd..16aa1d2 100644
--- feeds/luci/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
+++ feeds/luci/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
@@ -13,6 +13,10 @@ var callSystemInfo = rpc.declare({
 	method: 'info'
 });
 
+var callCPUBench = rpc.declare({
+	object: 'luci',
+	method: 'getCPUBench'
+});
 return baseclass.extend({
 	title: _('System'),
 
@@ -20,6 +24,7 @@ return baseclass.extend({
 		return Promise.all([
 			L.resolveDefault(callSystemBoard(), {}),
 			L.resolveDefault(callSystemInfo(), {}),
+			L.resolveDefault(callCPUInfo(), {}),
 			fs.lines('/usr/lib/lua/luci/version.lua')
 		]);
 	},
@@ -27,7 +32,8 @@ return baseclass.extend({
 	render: function(data) {
 		var boardinfo   = data[0],
 		    systeminfo  = data[1],
-		    luciversion = data[2];
+		    cpubench    = data[2],
+		    luciversion = data[3];
 
 		luciversion = luciversion.filter(function(l) {
 			return l.match(/^\s*(luciname|luciversion)\s*=/);
@@ -52,9 +58,9 @@ return baseclass.extend({
 
 		var fields = [
 			_('Hostname'),         boardinfo.hostname,
 			_('Model'),            boardinfo.model,
-			_('Architecture'),     boardinfo.system,
+			_('Architecture'),     cpuinfo.cpuinfo,
 			_('Firmware Version'), (L.isObject(boardinfo.release) ? boardinfo.release.description + ' / ' : '') + (luciversion || '') + ' built from Mr.D',
 			_('Kernel Version'),   boardinfo.kernel,
 			_('Local Time'),       datestr,
 			_('Uptime'),           systeminfo.uptime ? '%t'.format(systeminfo.uptime) : null,
-- 
2.14.1.windows.1
